modules/core/src/test/java/org/apache/ignite/testframework/junits/logger/GridTestLog4jLogger.java

```
@@ -474,6 +478,8 @@ public static void removeAppender(FileAppender a) {
         if (!impl.isInfoEnabled())
             warning("Logging at INFO level without checking if INFO level is enabled: " + msg);
 
+        assert impl.isInfoEnabled() : "Logging at INFO level without checking if INFO level is enabled: " + msg;
```

I think you should not throw an exception

---

Why?

---

If INFO is not enabled, you should throw an error so that when you disable the "ea" option, you do not miss the error.

---

"-ea" is enabled on TeamCity runs and IDE test configuration. Anyway according to your logic user can set other logger implementation and avoid those checks.